### PR TITLE
feat:762 Deployment Version Updating validation

### PR DESCRIPTION
**Pull request by Ropes**
* ~~Performance improvements to Active Lease lookup~~
* Code Documentation in the `provider` codebase
* Validation of manifest version to blockchain Deployment version
  * Asserts that manifest object provided is correct.
* ManifestVersion() method to extract Version from Manifest type.
* Makefile `deployment-update` command added
* Document Deployment update and (re)sending manifest.

fixes: #762

### CODE REVIEW

**Comment by Ropes** [comment]
Codecov isn't happy because no tests were added to the small-ish piece which does the validation check. 

Question to discuss: how should we test provider functionality? `integration/` tests don't go far enough to actually validate the provider's executing functionality. Extending the `TestSuite` would enable sharring the data stores between separate components.

File: client/extra.go [hunk 1]
```diff
@@ -6,14 +6,22 @@ import (
...
 
-// TODO: implement with search parameters
-
+// ActiveLeasesForProvider filter the Lease search to the provider's ID AccAddress
+// paramenter and 'active' state.
 func (c *qclient) ActiveLeasesForProvider(id sdk.AccAddress) (mquery.Leases, error) {
-	leases, err := c.Leases(mquery.LeaseFilters{})
+	// Establish Provider's search
+	lf := mquery.LeaseFilters{
+		Owner:        id,
```
[the remark this answers is not in the document]
Comment (Ropes) [reply client/extra.go]: :grimacing: yeah... so this is the filter struct:
```go
// LeaseFilters defines flags for lease list filter
type LeaseFilters struct {
	Owner sdk.AccAddress
	// State flag value given
	StateFlagVal string
	// Actual state value decoded from LeaseStateMap
	State types.LeaseState
}
```

After diving into query code, looks like we don't have the actual `Provider` address filtering functionality built in yet. However I'm also rather unsure how this raw ABCI query is actually being interpreted. I'd like to get some tests around it.. however if we want to get the version fix out, I reverted this change and we can get the Version update out.

[the remark this answers is not in the document]
Comment (boz) [reply client/extra.go]: right.  it's not ideal.  it shouldn't be using a string value, etc...

all of the ids are additive and the `Owner` field comes from the `DeploymentID`

File: client/client.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ type QueryClient interface {
 	pquery.Client
 
 	// TODO: implement with search parameters
-	ActiveLeasesForProvider(id sdk.AccAddress) (mquery.Leases, error)
+	ActiveLeasesForProvider(provider sdk.AccAddress) (mquery.Leases, error)
```
Comment (boz) [inline client/client.go:34]: eh

File: provider/manifest/manager.go [hunk 3]
```diff
@@ -301,7 +308,16 @@ func (m *manager) validateRequests() {
 }
 
 func (m *manager) validateRequest(req manifestRequest) error {
-	// TODO: hash(manifest) == m.data.Version
+	// ensure that an uploaded manifest matches the hash declared on
+	// the Akash Deployment.Version
+	manifestVersion, err := sdl.ManifestVersion(req.value.Manifest)
```
Comment (boz) [inline provider/manifest/manager.go:313]: :nail_care: too verbose.  `version` is fine.

File: client/extra.go [hunk 4]
```diff
@@ -6,17 +6,22 @@ import (
...
+	leases, err := c.Leases(lf)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: Secondary filtering *should* not be necessary
 	var filtered mquery.Leases
 	for _, lease := range leases {
-		if lease.Provider.Equals(id) && lease.State == types.LeaseActive {
+		if lease.Provider.Equals(provider) && lease.State == types.LeaseActive {
```
Comment (boz) [inline client/extra.go]: just leave this stuff as it was - these changes don't really add anything.

/bump #725 #604

File: provider/cluster/service.go [hunk 5]
```diff
@@ -212,16 +212,14 @@ loop:
...
-
 			s.log.Debug("manager done", "lease", dm.lease)
 
+			// unreserve resources
 			if _, err := s.inventory.unreserve(dm.lease.OrderID(), dm.mgroup); err != nil {
 				s.log.Error("unreserving inventory", "err", err,
 					"lease", dm.lease, "group-name", dm.mgroup.Name)
 			}
 
-			// todo: unreserve resources
```
Comment (boz) [inline provider/cluster/service.go:223]: :+1:

File: client/extra.go [hunk 6]
```diff
@@ -6,14 +6,19 @@ import (
...
+	// NOTE: Currently unable to filter by Provider in tendermint scan
+	lf := mquery.LeaseFilters{
+		// TODO: Add filters when tests prove functionality
+	}
+	leases, err := c.Leases(lf)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: Secondary filtering *should* not be necessary
```
Comment (boz) [inline client/extra.go:21]: Just leave this stuff as it was before - these changes don't really add anything.

refs #604 #725

Comment (Ropes) [reply client/extra.go]: :heavy_check_mark:

**Pull request merged**